File: onebot-client/src/main/java/cn/evole/onebot/sdk/action/ActionData.java
package cn.evole.onebot.sdk.action;


import com.google.gson.annotations.SerializedName;
import lombok.Data;

/**
 * Description:
 * Author: cnlimiter
 * Date: 2022/9/13 22:05
 * Version: 1.0
 */
@Data
public class ActionData<T> {

    @SerializedName("status")
    private String status;
    @SerializedName("retcode")
    private int retCode;
    @SerializedName("data")
    private T data;
    @SerializedName("echo")
    private String echo;
}

File: onebot/src/main/kotlin/sdk/action/ActionData.kt
package cn.evolvefield.onebot.sdk.action

import com.google.gson.annotations.SerializedName
import lombok.Data

@Data
class ActionData<T : Any> {
    @SerializedName("status")
    var status = "failed"
    @SerializedName("retcode")
    var retCode = 1400
    @SerializedName("data")
    var data: T? = null
    @SerializedName("echo")
    var echo: String? = null
}
